tests/test_one_client_read_loop_server.py

```diff
@@ -0,0 +1,84 @@
+# This file is part of ts_tcpip.
+#
+# Developed for the Rubin Observatory Telescope and Site System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import logging
+import unittest
+
+from lsst.ts import tcpip
+
+# Standard timeout in seconds.
+TIMEOUT = 60
+
+
+class OneClientReadLoopServerTestCase(unittest.IsolatedAsyncioTestCase):
+    async def asyncSetUp(self) -> None:
+        self.srv = tcpip.TestOneClientReadLoopServer()
+        self.log = logging.getLogger(type(self).__name__)
+
+        await self.srv.start_task
+        assert self.srv.server.is_serving()
+        self.reader, self.writer = await asyncio.open_connection(
+            host=self.srv.host, port=self.srv.port
+        )
+        while not self.srv.connected:
+            await asyncio.sleep(0.1)
+        assert self.srv.connected
+
+    async def asyncTearDown(self) -> None:
+        if self.srv.connected:
+            await self.srv.disconnect()
+        if self.writer:
+            self.writer.close()
+            await self.writer.wait_closed()
+        await self.srv.close()
+
+    async def read(self) -> str:
+        """Read a string from the reader.
+
+        Returns
+        -------
+        message : `str`
+            The string read.
+        """
+        message = await asyncio.wait_for(
+            self.reader.readuntil(tcpip.TERMINATOR), timeout=TIMEOUT
+        )
+        return message.decode()
+
+    async def write(self, message: str) -> None:
+        """Write the message appended with a tcpip.TERMINATOR string.
+
+        Parameters
+        ----------
+        message : `str`
+            The message to write.
+        """
+        assert self.writer is not None
+        self.writer.write(message.encode() + tcpip.TERMINATOR)
+        await self.writer.drain()
+
+    async def test_read_and_dispatch(self) -> None:
+        test_message = "Test message."
+        await self.write(message=test_message)
```

Comment:
Several suggestions for beefing this up:
* Read and write several messages. That makes sure the read loop keeps running after parsing the first message.
* Test what the read loop does if read_and_dispatch raises. Two ways that come to mind to do that:
  * Add a flag "fail_next_read" to the test class which you reset just before raising
  * Make the test class read json messages, then write a message that is not valid json.